lib/std/std.zig

```diff
@@ -73,5 +73,10 @@ comptime {
 }
 
 test "" {
-    meta.refAllDecls(@This());
+    // TODO is there a way around this? When enabled for WASI, we pick up functions
+    // which generate compile error. Perhaps semantic analyser should skip those
+    // if running in test mode?
```

Comment:
Ah, makes sense! Thanks for explaining what `refAllDecls` is. So the reason I was skipping that was because otherwise the compiler was tripping on `@compileError` in a few places in `os.zig` and `fs.zig` such as `fn cwd()` which is indeed a compile error in WASI. Is there any way of telling the compiler to skip some functions in some namespace when cross-compiling to some target? I'm thinking here that if we could tell the compiler to skip fns such as `cwd()` or `getcwd()` that would make this check unnecessary.

Reply:
OK, I've now removed the check. It turns out I missed `lib/std/build.zig` when searching for `cwd()` and related uses, and since tests in `lib/std/build.zig` were not conditioned on whether we're cross-compiling to WASI or not, it was always pulling in the functions/structs that generate `@compileError` in WASI. So that's fixed now in 3d267ba.